src/core/platforms/android/androidplatformutilities.cpp

```diff
@@ -345,6 +346,10 @@ PictureSource *AndroidPlatformUtilities::getCameraPicture( QQuickItem *parent, c
   if ( !checkCameraPermissions() )
     return nullptr;
 
+  const QFileInfo destinationInfo( prefix + pictureFilePath );
+  const QDir prefixDir( prefix );
+  prefixDir.mkpath( destinationInfo.absolutePath() );
```

Comment:
Probably a good idea to check whether the `mkpath` returned true?

Reply:
I'll take a note of it here and think of how we can address that in a better way. On that front the old code also assumed the mkdirs() would always work. I think we'll need some handling on the QML side too.

In any case, I'd like to keep this limited to moving the mkdirs to a mkpath for now and bring the fix to 2.6.